lua/cspell/helpers.lua

```diff
@@ -11,44 +10,22 @@ local CSPELL_CONFIG_FILES = {
     ".cSpell.json",
     ".cspell.config.json",
 }
-
 ---@type table<string, CSpellConfigInfo|nil>
-local CONFIG_INFO_BY_CWD = {}
-local PATH_BY_CWD = {}
+local CONFIG_INFO_BY_PATH = {}
+---@type table<string, string|nil>
+local PATH_BY_DIRECTORY = {}
 
---- create a bare minimum cspell.json file
----@param params GeneratorParams
----@return CSpellConfigInfo
-M.create_cspell_json = function(params)
+local create_cspell_json = function(params, cspell_json, cspell_json_file_path)
     ---@type CSpellSourceConfig
     local code_action_config = params:get_config()
-    local config_file_preferred_name = code_action_config.config_file_preferred_name or "cspell.json"
     local encode_json = code_action_config.encode_json or vim.json.encode
-
-    if not vim.tbl_contains(CSPELL_CONFIG_FILES, config_file_preferred_name) then
-        vim.notify(
-            "Invalid config_file_preferred_name for cspell json file: "
-                .. config_file_preferred_name
-                .. '. The name "cspell.json" will be used instead',
-            vim.log.levels.WARN,
-            { title = "cspell.nvim" }
-        )
-        config_file_preferred_name = "cspell.json"
-    end
-
-    local cspell_json = {
-        version = "0.2",
-        language = "en",
-        words = {},
-        flagWords = {},
-    }
-
     local cspell_json_str = encode_json(cspell_json)
-    local cspell_json_file_path = require("null-ls.utils").path.join(params.cwd, config_file_preferred_name)
 
+    local cspell_json_directory_path = vim.fs.dirname(cspell_json_file_path)
+    Path:new(cspell_json_directory_path):mkdir({ parents = true })
     Path:new(cspell_json_file_path):write(cspell_json_str, "w")
     vim.notify(
-        "Created a new cspell.json file at " .. cspell_json_file_path,
+        "Created a new cspell.json file at " .. M.shorten_path(cspell_json_file_path),
         vim.log.levels.INFO,
         { title = "cspell.nvim" }
     )
```

Comment:
This message will now show up every time a user opens up a file in a directory new to the cache. That sounds like we should either remove this notification or delegate it to the none-ls logger with a debug level.

Reply:
Sounds good. Users can now check for a file creation messages in the none-ls logs if they change their log level to debug, or by enabling the debug mode in none-ls.